File: wrappers/dotnet/indy-sdk-dotnet-test/Wrapper/PoolTests/ClosePoolTest.cs
using Indy.Sdk.Dotnet.Wrapper;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Indy.Sdk.Dotnet.Test.Wrapper.PoolTests
{
    [TestClass]
    public class ClosePoolTest : IndyIntegrationTest
    {
        [TestMethod]
        public void TestClosePoolWorks()
        {
            var pool = PoolUtils.CreateAndOpenPoolLedger();
            Assert.IsNotNull(pool);
            _openedPools.Add(pool);

            pool.CloseAsync().Wait();
            _openedPools.Remove(pool);
        }

        [TestMethod]
        public async Task TestClosePoolWorksForTwice()
        {
            var pool = PoolUtils.CreateAndOpenPoolLedger();
            Assert.IsNotNull(pool);
            _openedPools.Add(pool);

            pool.CloseAsync().Wait();
            _openedPools.Remove(pool);

            var ex = await Assert.ThrowsExceptionAsync<IndyException>(() =>
                pool.CloseAsync()
            );

            Assert.AreEqual(ErrorCode.PoolLedgerInvalidPoolHandle, ex.ErrorCode);
        }

        [TestMethod]
        public void TestClosePoolWorksForReopenAfterClose()
        {
            var poolName = PoolUtils.CreatePoolLedgerConfig();
            var pool = Pool.OpenPoolLedgerAsync(poolName, null).Result;

            Assert.IsNotNull(pool);

            pool.CloseAsync().Wait();

            pool = Pool.OpenPoolLedgerAsync(poolName, null).Result;
            _openedPools.Add(pool);
        }
    }
}

File: wrappers/android/src/androidTest/java/org/hyperledger/indy/sdk/pool/ClosePoolTest.java
package org.hyperledger.indy.sdk.pool;

import org.hyperledger.indy.sdk.IndyIntegrationTest;
import org.hyperledger.indy.sdk.utils.PoolUtils;
import org.junit.Test;

import static org.junit.Assert.assertNotNull;

public class ClosePoolTest extends IndyIntegrationTest {

	@Test
	public void testClosePoolWorks() throws Exception {
		Pool pool = PoolUtils.createAndOpenPoolLedger();
		assertNotNull(pool);
		openedPools.add(pool);

		pool.closePoolLedger().get();
		openedPools.remove(pool);
	}


	@Test
	public void testAutoCloseWorks() throws Exception {
		String poolName = PoolUtils.createPoolLedgerConfig();
		try (Pool pool = Pool.openPoolLedger(poolName, null).get()) {
			assertNotNull(pool);
		}
		Pool pool = Pool.openPoolLedger(poolName, null).get();
		openedPools.add(pool);
	}
}
